fix splunk max buffer size (#227)
centreon/centreon-stream-connector-scripts: modules/docs/sc_common.md

local result = test_common:is_valid_pattern(wrong_pattern)
--> result is: false
```

## sleep method

> The **sleep** method make the script pause for the given amount of seconds.

Reviewer: ```suggestion
The **sleep** method makes the script pause for the given number of seconds.
```

### sleep: parameters

| parameter                              | type   | optional | default value |
| -------------------------------------- | ------ | -------- | ------------- |
| the number of seconds you need to wait | number | no       |               |

### sleep: example

```lua
local seconds_to_wait = 10

test_common:sleep(seconds_to_wait)
```

## create_sleep_counter_table method

The **create_sleep_counter_table** method creates a table to handle sleep counters. Useful when you want to log something less often after a few times.

> This table will contain a min, a max, a value and a step number that will be used to increment the value. The table is also shipped with three methods

Reviewer: ```suggestion
This table will contain a min, a max, a value and a step number that will be used to increment the value. The table is also shipped with three methods.
```

| method name      | purpose                                                                                                                          | params | returns                                                   |
| ---------------- | -------------------------------------------------------------------------------------------------------------------------------- | ------ | --------------------------------------------------------- |
| reset()          | set the counter value to the minimum value                                                                                       |        |                                                           |

> | is_max_reached() | checks if the counter is equal or above the max                                                                                  |        | true or false if the counter is equal or above to the max |

Reviewer: ```suggestion
| is_max_reached() | checks if the counter is equal to or above the max                                                                                  |        | true or false if the counter is equal to or above the max |
```

> | increment()      | increase the counter by adding to it the step number                                                                             |        |                                                           |

Reviewer: ```suggestion
| increment()      | increase the counter by adding the step number to it                                                                           |        |                                                           |
```

> | sleep()          | will use the sc_common:sleep() function using the counter value as an argument and it will automatically increment() the counter |        |                                                           |

Reviewer: ```suggestion
| sleep()          | will use the sc_common:sleep() function using the counter value as an argument, it will automatically increment() the counter |        |                                                           |
```

> here is an example of a log that is written less and less often thanks to this function (min was 0, max: 300, step: 10)

Reviewer: ```suggestion
Here is an example of a log that is written less and less often thanks to this function (min was 0, max: 300, step: 10).
```

Tue Aug 13 16:17:14 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:17:34 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:18:04 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:18:44 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:19:34 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:20:34 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:21:44 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:23:04 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:24:34 2024: ERROR: Skipping event because a mandatory parameter is not set
Tue Aug 13 16:26:14 2024: ERROR: Skipping event because a mandatory parameter is not set
```

### create_sleep_counter_table: parameters

| parameter                                                                | type   | optional | default value |
| ------------------------------------------------------------------------ | ------ | -------- | ------------- |

> | an empty table that will be returned with all the desired data structure | table  | no       |               |

Reviewer: ```suggestion
| an empty table that will be returned with the desired data structure | table  | no       |               |
```
